### PR TITLE
Support for G-Drive Integration 

### PR DESCRIPTION
**Pull request by balrampariyarath**

### CODE REVIEW

File: lib/rucio/rse/protocols/gdrive.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import os
+from pydrive.auth import GoogleAuth
+from pydrive.drive import GoogleDrive
+
+from rucio.common.utils import execute
+from rucio.common import exception
+from rucio.rse.protocols import protocol
```
Comment (vingar) [inline lib/rucio/rse/protocols/gdrive.py]: order import in the alphabetical ordered
First system ones

then third parties ones

File: lib/rucio/rse/protocols/gdrive.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import os
```
Comment (vingar) [inline lib/rucio/rse/protocols/gdrive.py]: comments and headers missing

File: etc/rse-accounts.cfg.template [hunk 3]
```diff
@@ -21,5 +21,15 @@
...
     "auth_type": "cert",
     "timeout":300
+  },
+  "GOOGLE-DRIVE_RSE-OAUTH": {
+    "client_id": "628137923541-8hna5qlb3bvvm2f9m5bius6o1avlsesi.apps.googleusercontent.com",
+    "project_id": "rucio-171207",
+    "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+    "token_uri": "https://accounts.google.com/o/oauth2/token",
+    "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+    "client_secret": "YtgihX8M9vcDrYBrXjRxXmCV",
```
Comment (vingar) [inline etc/rse-accounts.cfg.template]: remove the secret keys...

File: lib/rucio/tests/test_rse_gdrive.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import rucio.rse.protocols.gdrive
```
Comment (vingar) [inline lib/rucio/tests/test_rse_gdrive.py]: comments missing

File: lib/rucio/rse/protocols/gdrive.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+		gauth.LocalWebserverAuth() # Creates local webserver and auto handles authentication
+    	return GoogleDrive(gauth)
+
+    def upload_to_drive(self, filename):
+    	filename = filename + ".txt"
+		file = self.drive.CreateFile({'title': filename})
+		file.Upload()
+		return file['id']
+
+	def create_subfolder(self, folder, sfldname):
```
Comment (mlassnig) [inline lib/rucio/rse/protocols/gdrive.py]: bad indentation

File: lib/rucio/rse/protocols/gdrive.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    	new_folder = self.drive.CreateFile({
+    		'title':'{}'.format(sfldname),
+    		'mimeType':'application/vnd.google-apps.folder'
+    	})
+    	if folder is not None:
+    	    new_folder['parents'] = [{u'id': folder['id']}]
+    	new_folder.Upload()
+		return new_folder
+
+	def upload_files_to_folder(self, fnames, folder):
```
Comment (mlassnig) [inline lib/rucio/rse/protocols/gdrive.py]: bad indentation

File: lib/rucio/rse/protocols/gdrive.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+		for fname in fnames: 
+        	nfile = self.drive.CreateFile({
+        		'title':os.path.basename(fname),
+        		'parents':[{u'id': folder['id']}]
+        	})
+        	nfile.SetContentFile(fname)
+			nfile.Upload() 
+
+	def test(self):
```
Comment (mlassnig) [inline lib/rucio/rse/protocols/gdrive.py]: bad indentation

File: lib/rucio/tests/test_rse_gdrive.py [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import rucio.rse.protocols.gdrive 
+
+
+class TestDriveRSE:
+
+	def test_upload(self, gdrive):
```
Comment (mlassnig) [inline lib/rucio/tests/test_rse_gdrive.py]: bad indentation

File: lib/rucio/rse/protocols/gdrive.py [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+from rucio.rse.protocols import protocol
+
+
+class Default(protocol.RSEProtocol):
+    """ Implementing RSE using GDrive protocol. """
+
+    @property
+    def drive(self, credentials):
+    	gauth = GoogleAuth()
+		gauth.LocalWebserverAuth() # Creates local webserver and auto handles authentication
```
Comment (mlassnig) [inline lib/rucio/rse/protocols/gdrive.py]: bad indentation

**Comment by vingar** [comment]
please run flake8 and pylint on the code before committing it

File: lib/rucio/rse/protocols/gdrive.py [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+    	return GoogleDrive(gauth)
+
+    def upload_to_drive(self, filename):
+    	filename = filename + ".txt"
+		file = self.drive.CreateFile({'title': filename})
```
Comment (mlassnig) [inline lib/rucio/rse/protocols/gdrive.py]: ok, i'm stopping with the bad indentation comments now.

File: lib/rucio/rse/protocols/gdrive.py (hunk 1 above)
Comment (balrampariyarath) [inline lib/rucio/rse/protocols/gdrive.py]: ok

File: lib/rucio/rse/protocols/gdrive.py (hunk 10 above)
Comment (balrampariyarath) [inline lib/rucio/rse/protocols/gdrive.py]: sorry about that..will fix it..

**Comment by balrampariyarath** [comment]
I had made the changes. The code Connects with G-Drive instance using PyDrive wrapper. More tests needs to be added, I am working on it.

**Pull request closed without merging**